x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/TextFormat.java

```diff
@@ -294,7 +295,11 @@ public String subtype() {
     private static final String PARAM_HEADER_ABSENT = "absent";
     private static final String PARAM_HEADER_PRESENT = "present";
 
-    private static final MediaTypeParser<TextFormat> parser = new MediaTypeParser<>(TextFormat.values());
+    private static final MediaTypeParser<TextFormat> parser = new MediaTypeParser.Builder<TextFormat>()
+        .withMediaTypeAndParams(PLAIN_TEXT.typeWithSubtype(), PLAIN_TEXT, Map.of("header", "present|absent", "charset", "utf-8"))
```

Comment:
I am concerned that having charset with a regex of utf-8 could break someone using sql. This should be checked with the sql team

Reply:
this only means that a parameter is possible. if it won't be provided it won't fail any validation.
I asked SQL team about this offline.
 @bpintea  can you confirm that we can expect any of these in SQL? Do you expect any other charsets?

Reply:
Not speaking for SQL, but in general with a lot of clients on Windows, I have seen them send `charset=ISO-8859-1` with the `Content-Type` header and would operate under the assumption that they may also send the charset with the `Accept` header; nothing in RFC 7231 would prevent that. Additionally, multiple accept headers are allowed so you can wind up with something like:

```
Accept: text/plain;charset="utf-8", text/plain;charset="ISO-8859-1"
```

There is also the `Accept-Charset` header which technically should have precedence over the media type parameter in the `Accept` header. (It is kind of a mess)

Reply:
I think having multiple headers would be beyond the scope of this change. We would also have to support the `q` factor to understand the preference of clients Accept type.

Reply:
> if it won't be provided it won't fail any validation.

Right, I believe currently the `TextFormat`er simply ignores an `Accept`-requested charset (that happens to be different than UTF-8). Hasn't been an issue so far, but it might make sense to do something about it at some point.